src/main/java/seedu/recipe/ui/events/EditRecipeEvent.java

```diff
@@ -0,0 +1,28 @@
+package seedu.recipe.ui.events;
+
+import java.util.Map;
+
+import javafx.event.Event;
+import javafx.event.EventType;
+
+public class EditRecipeEvent extends Event {
+    public static final EventType<EditRecipeEvent> EDIT_RECIPE_EVENT_TYPE = new EventType<>(Event.ANY, "EDIT_RECIPE");
+
+    private final int recipeIndex;
+
+    private final Map<String, String> changedValues;
```

Comment:
Can reference `ParserUtil.java` to see how they handle various fields based off a singular command string

Reply:
It's basically using `ArgumentMultiMap` which is a glorified `HashMap`, but with recipe field-specific hash keys using `Prefix`